repos/system_upgrade/common/libraries/overlaygen.py

```diff
@@ -42,39 +137,74 @@ def _mount_name(mountpoint):
 def _mount_dir(mounts_dir, mountpoint):
     return os.path.join(mounts_dir, _mount_name(mountpoint))
 
+def _get_mountpoint(mount_points, dir_path):
+    for mp in sorted(mount_points, reverse=True):
+        mod_mp = mp if mp[-1] == '/' else '{}/'.format(mp)
+        if dir_path.startswith(mod_mp):
+            # longest first, so the first one we find, is the last mp on the path
+            return mp
 
-def _prepare_required_mounts(scratch_dir, mounts_dir, mount_points, xfs_info):
-    result = {
-        mount_point.fs_file: mounting.NullMount(
-            _mount_dir(mounts_dir, mount_point.fs_file)) for mount_point in mount_points
-    }
-
-    if not xfs_info.mountpoints_without_ftype:
-        return result
 
-    space_needed = _overlay_disk_size() * len(xfs_info.mountpoints_without_ftype)
+def _prepare_required_mounts(scratch_dir, mounts_dir, storage_info, scratch_reserve):
+    """
+    Create disk images and loop mount them.
+
+    Ensure to create disk image for each important mountpoint configured
+    in fstab (excluding fs types noted in `OVERLAY_DO_NOT_MOUNT`).
+    Disk images reflect the free space of related partition/volume. In case
+    of partition hosting /var/lib/leapp/* calculate the free space value
+    taking `scratch_reserve` into account, as during the run of the tooling,
+    we will be consuming the space on the partition and we want to be more
+    sure that we do not consume all the space on the partition during the
+    execution - so we reduce the risk we affect run of other applications
+    due to missing space.
+
+    Note: the partition hosting the scratch dir is expected to be the same
+    partition that is hosting the target userspace container, but it does not
+    have to be true if the code changes. Right now, let's live with that.
```

Comment:
Just out of curiosity, do we have any plans for the code changes?

Reply:
Question is not whether yes or no. Important is to be aware that we have some functionality implemented in the way that if a change happens, this will be broken with the current implementation. No, we do not plan changes here. But who knows in years?